src/axes/axis.ts

```diff
@@ -105,6 +105,27 @@ export class Axis<D> extends Component {
     this._scale.offUpdate(this._rescaleCallback);
   }
 
+  /**
+   * Gets the tick label data on a element.
+   *
+   * @param {Element} element
+   */
+  public tickLabelDataOnElement(element: Element) {
+    if (element === null || element === undefined) { return undefined; }
+
+    let tickLabel: Element;
+    // go up DOM tree to find tick label element in ancestor elements
+    while ((element !== undefined) && (element.classList) && (tickLabel === undefined)) {
+      if (element.classList.contains(Axis.TICK_LABEL_CLASS)) {
+        tickLabel = element;
+      } else {
+        element = element.parentNode as Element;
```

Comment:
it's fine because we do the check for `element` and `classList` in the `while` statement

> parentNode is the parent of the current node. The parent of an element is an Element node, a Document node, or a DocumentFragment node.

if it gets to a Document node we want to stop, we want to stop the search